core-api/src/main/java/org/silverpeas/core/variables/VariableValueSet.java

```diff
@@ -0,0 +1,240 @@
+package org.silverpeas.core.variables;
+
+import org.silverpeas.core.persistence.Transaction;
+
+import java.time.LocalDate;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A set of variable's values. The set is related to a given variable and each value is defined for
+ * a given period in time.
+ * @author mmoquillon
+ */
+public class VariableValueSet implements Collection<VariableScheduledValue> {
+
+  final Variable variable;
+
+  VariableValueSet(final Variable variable) {
+    this.variable = variable;
+  }
+
+  /**
+   * Adds the specified variable value into this variable's values set. In order the adding to
+   * be effective you have to invoke {@link Variable#save()}.
+   * @param value the value, defined in a given period, to add in this set.
+   */
+  public boolean add(final VariableScheduledValue value) {
+    value.setVariable(variable);
+    return variable.values().add(value);
+  }
+
+  @Override
+  public boolean remove(final Object o) {
+    return false;
+  }
+
+  @Override
+  public boolean containsAll(final Collection<?> c) {
+    return false;
+  }
+
+  /**
+   * Adds the specified variable values into this variable's values set. In order the adding to
+   * be effective you have to invoke {@link Variable#save()}.
+   * @param values a collection of values, each of then defined in a given period, to add in this
+   * set.
+   */
+  public boolean addAll(final Collection<? extends VariableScheduledValue> values) {
+    return variable.values()
+        .addAll(values.stream().peek(v -> v.setVariable(variable)).collect(Collectors.toList()));
+  }
+
+  @Override
+  public boolean removeAll(final Collection<?> c) {
+    return variable.values().removeAll(c);
+  }
+
+  @Override
+  public boolean retainAll(final Collection<?> c) {
+    return variable.values().retainAll(c);
+  }
+
+  @Override
+  public void clear() {
+    variable.values().clear();
+  }
+
+  /**
+   * Adds the specified variable value into the variable's values set and saves automatically the
+   * set. This method can be invoked only with an already persisted variable otherwise a
+   * {@link javax.persistence.PersistenceException} will be thrown. Useful when just to add a new
+   * value to the underlying variable for a given period.
+   * @param value the value, defined in a given period, to add in this set.
+   * @return the persisted variable value.
+   */
+  public VariableScheduledValue addAndSave(final VariableScheduledValue value) {
+    return Transaction.performInOne(() -> {
+      value.setVariable(variable);
+      return VariableScheduledValueRepository.get().save(value);
+    });
+  }
+
+  /**
+   * Gets a stream on all of the values of the underlying variable.
+   * @return a stream of variable values.
+   */
+  public Stream<VariableScheduledValue> stream() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1161)